src/tests/ftest/erasurecode/restart.py

```diff
@@ -31,43 +31,81 @@ def execution(self, agg_check=None):
                        after.Default is None so not to check aggregation, but wait for 20 seconds
                        and restart the servers.
         """
-        # Write all EC object data to NVMe
+        # 1.
+        aggr_threshold = self.params.get("aggregation_threshold", '/run/ior/*', default=300)
+        self.log_step("Create pool")
+
+        # 2.
+        self.log_step("Disable aggregation")
+        self.pool.set_property("reclaim", "disabled")
+        # Get initial total free space (scm+nvme)
+        initial_free_space = self.pool.get_total_free_space(refresh=True)
+        self.log.info("initial pool free space: %s", initial_free_space)
+
+        # 3.
+        self.log_step("Run IOR write all EC object data to container")
         self.ior_write_dataset(operation="Auto_Write", percent=self.percent)
-        self.log.info(self.pool.pool_percentage_used())
-        # Write all EC object data to SCM
+        free_space_after_ior = self.pool.get_total_free_space(refresh=True)
+        self.log.info("pool free space after IOR write: %s", free_space_after_ior)
+        self.assertTrue(free_space_after_ior < initial_free_space, "IOR run was not successful.")
+
+        if agg_check == "Restart_before_agg":
+            # step-4 for Restart_before_agg test
+            self.log_step("Shutdown the servers and restart")
+            self.get_dmg_command().system_stop(True)
+            self.get_dmg_command().system_start()
```

Comment:
Let's check for aggregation before restart - if set - and then always restart.  If aggregation :
```suggestion
        self.log_step("Enable aggregation")
        self.pool.set_property("reclaim", "time")

        if agg_check == "Restart_after_agg":
           self.log_step("Verify aggregation is complete before engine restart")
            self.verify_aggreation()
        
        self.log_step("Stop the engines (dmg system stop)")
        self.get_dmg_command().system_stop(True)
        self.log_step("Restart the engines (dmg system start)")
        self.get_dmg_command().system_start()
        
        if agg_check == "Restart_before_agg":
            self.log_step("Verify aggregation is complete after engine restart")
            self.verify_aggreation()
            
        self.log_step("Verify data after aggregation (ior read)")
        ior_kwargs['flags'] = ior_read_flags
        for container in containers:
            ior_kwargs['container'] = container
            try:
                result = run_ior(**ior_kwargs)
            except CommandFailure as error:
                self.fail('ior failed')
                
        self.log_step("Test passed")
```

Where `self.verify_aggreation()` looks something like:
```
  def verify_aggreation(self, first_ior_free_space):
        self.log.info("Waiting for aggregation to complete")
        if not wait_for_result(self.log, 150, check_free_space_equals, expected=first_ior_free_space):
            self.fail("aggregation completion not detected")

  def check_free_space_equals(self, expected):
        return expected == self.pool.get_total_free_space(refresh=True)
```